domain-models-api-aspects/src/main/java/org/folio/rest/validators/CollectionValidator.java

```diff
@@ -0,0 +1,14 @@
+package org.folio.rest.validators;
+
+import java.lang.annotation.Annotation;
+import java.util.Collection;
+import java.util.Optional;
+
+import javax.validation.ConstraintValidator;
+
+public interface CollectionValidator<A extends Annotation, E, C extends Collection<E>>
+    extends ConstraintValidator<A, C> {
+
+  public boolean isEntityValid(Optional<E> element);
```

Comment:
What is "entity"? Or can you rename "isEntityValid" to "isValid"?

Reply:
Probably should be isElementValid. The method is being invoked within an iteration upon the elements of an array. Would not want to confuse the validation of the entire collection with the validation of a single element within the collection.